src/stacks-hierarchy/Root_PurchaseOverviewScreen/Root_PurchaseOverviewScreen.tsx

```diff
@@ -277,16 +277,27 @@ export const Root_PurchaseOverviewScreen: React.FC<Props> = ({
                 travelDate,
                 mode: params.mode,
               });
-              navigation.navigate('Root_PurchaseConfirmationScreen', {
-                fareProductTypeConfig: params.fareProductTypeConfig,
-                fromPlace: fromPlace,
-                toPlace: toPlace,
-                userProfilesWithCount: travellerSelection,
-                preassignedFareProduct,
-                travelDate,
-                headerLeftButton: {type: 'back'},
-                mode: params.mode,
-              });
+              isOnBehalfOfToggle ?
+                navigation.navigate('Root_ChooseTicketReceiverScreen', {
+                  fareProductTypeConfig: params.fareProductTypeConfig,
+                  fromPlace: fromPlace,
+                  toPlace: toPlace,
+                  userProfilesWithCount: travellerSelection,
+                  preassignedFareProduct,
+                  travelDate,
+                  headerLeftButton: {type: 'back'},
+                  mode: params.mode,
+                }) :
+                navigation.navigate('Root_PurchaseConfirmationScreen', {
+                  fareProductTypeConfig: params.fareProductTypeConfig,
+                  fromPlace: fromPlace,
+                  toPlace: toPlace,
+                  userProfilesWithCount: travellerSelection,
+                  preassignedFareProduct,
+                  travelDate,
+                  headerLeftButton: {type: 'back'},
+                  mode: params.mode,
+                });
```

Comment:
If you first define
```ts
const rootPurchaseConfirmationScreenParams = {
    fareProductTypeConfig: params.fareProductTypeConfig,
    fromPlace: fromPlace,
    toPlace: toPlace,
    userProfilesWithCount: travellerSelection,
    preassignedFareProduct,
    travelDate,
    headerLeftButton: {type: 'back'},
    mode: params.mode,
}
```

Then you can simplify this by sending `{rootPurchaseConfirmationScreenParams}` and `rootPurchaseConfirmationScreenParams` to the navigate calls accordingly

Reply:
done